R/as_gt.R

```diff
@@ -80,105 +80,44 @@ as_gt <- function(x, ...) {
 #'   as_gt()
 as_gt.fixed_design <- function(x, title = NULL, footnote = NULL, ...) {
   # get the design method
-  if ("ahr" %in% class(x)) {
-    design_mtd <- "ahr"
-  } else if ("fh" %in% class(x)) {
-    design_mtd <- "fh"
-  } else if ("mb" %in% class(x)) {
-    design_mtd <- "mb"
-  } else if ("lf" %in% class(x)) {
-    design_mtd <- "lf"
-  } else if ("rd" %in% class(x)) {
-    design_mtd <- "rd"
-  } else if ("maxcombo" %in% class(x)) {
-    design_mtd <- "maxcombo"
-  } else if ("milestone" %in% class(x)) {
-    design_mtd <- "milestone"
-  } else if ("rmst" %in% class(x)) {
-    design_mtd <- "rmst"
-  } else if ("rd" %in% class(x)) {
-    design_mtd <- "rd"
-  }
-
+  design_mtd <- intersect(
+    c("ahr", "fh", "mb", "lf", "rd", "maxcombo", "milestone", "rmst"), class(x)
+  )[1]
 
   # set the default title
-  if (is.null(title)) {
-    title <- switch(design_mtd,
-      "ahr" = {
-        "Fixed Design under AHR Method"
-      },
-      "fh" = {
-        "Fixed Design under Fleming-Harrington Method"
-      },
-      "mb" = {
-        "Fixed Design under Magirr-Burman Method"
-      },
-      "lf" = {
-        "Fixed Design under Lachin and Foulkes Method"
-      },
-      "rd" = {
-        "Fixed Design of Risk Difference under Farrington-Manning Method"
-      },
-      "maxcombo" = {
-        "Fixed Design under MaxCombo Method"
-      },
-      "milestone" = {
-        "Fixed Design under Milestone Method"
-      },
-      "rmst" = {
-        "Fixed Design under Restricted Mean Survival Time Method"
-      },
-      "rd" = {
-        "Fixed Design of Risk Difference"
-      }
-    )
-  }
-
+  if (is.null(title)) title <- sprintf("Fixed Design under %s Method", switch(
+    design_mtd,
+    ahr = "AHR", fh = "Fleming-Harrington", mb = "Magirr-Burman",
+    lf = "Lachin and Foulkes", maxcombo = "MaxCombo", milestone = "Milestone",
+    rd = "Fixed Design of Risk Difference under Farrington-Manning",
+    rmst = "Restricted Mean Survival Time"
+  ))
```

Comment:
This isn't equivalent to the `switch()` statement that was replaced, for example when `design_mtd <- "rd"`

```R
# before
[1] "Fixed Design of Risk Difference under Farrington-Manning Method"
# after
[1] "Fixed Design under Fixed Design of Risk Difference under Farrington-Manning Method"
```

Reply:
Good catch. Fixed now. Thanks!